lib/cocoapods/target/pod_target.rb

```diff
@@ -290,7 +290,7 @@ def recursive_dependent_targets
     #         The name of the build configuration.
     #
     def include_in_build_config?(target_definition, configuration_name)
-      key = [target_definition.label, configuration_name]
+      key = [target_definition, configuration_name]
```

Comment:
Caching still works. `target_definition.rb` does not implement `hash` and `eql` but it works due to the fact that the target definition instances are the same.